### DISCUSSION TITLE
Change Plot Question

### DISCUSSION THREAD

**Discussion by Alex-Wenner-FHR** [h2oai/wave#873] [Q&A]
Trying to change a specific point on the plot conditionally, using an update that looks like:

q.page['trend'].items[0].visualization.plot.mark.color= '$red'

but having no luck. Is this the right way to change the parameters of a mark component?

The structure of the plot is ui.form_card -> items -> ui.visualization -> ui.plot -> ui.mark

**Comment by vopani** [comment]
@AlexWennerDT Can you try this:
`q.page['trend'].items[0].visualization.plot.marks[0].color = '$red'`

The argument inside `plot` is `marks` and since it's a list, you need to choose the index of the mark to re-color.

**Reply by Alex-Wenner-FHR** [reply]
Thank you that worked!

**Comment by Alex-Wenner-FHR** [comment]
Hi! That line worked, but I am trying to conditionally change the plot points. I have this snippet of code: 
![image](https://user-images.githubusercontent.com/78372056/121924680-bfb67800-cd01-11eb-8a78-2aea5f1fb71a.png)


I would have expected it to only turn the points of data that are in the .loc. However it turns all of the points on my graph red, any thoughts?
![image](https://user-images.githubusercontent.com/78372056/121924607-ae6d6b80-cd01-11eb-922d-8010f8948960.png)

**Reply by vopani** [reply]
A better / cleaner approach might be to use `color_domain` and `color_range`.

**Reply by Alex-Wenner-FHR** [reply]
Hm, I was looking into that but was unsure what that would look like with a dataframe(s)

**Reply by vopani** [reply]
Try this:

```python
import pandas as pd

from h2o_wave import site, data, ui

page = site['/demo']

df = pd.DataFrame({
    'toy': ['X0', 'X1', 'X2', 'X3', 'X4'],
    'height': [1, 2, 3, 4, 8],
    'coloring': ['red', 'green', 'red', 'green', 'blue']
})


page.add('example', ui.form_card(
    box='1 1 4 5',
    items=[
        ui.text_xl('Example'),
        ui.visualization(
            plot=ui.plot([ui.mark(type='interval', x='=toy', y='=height', color='=coloring',
                                  color_domain=['blue', 'green', 'red'], color_range='blue green red')]),
            data=data(fields='toy height coloring', columns=[df[col].to_list() for col in df.columns], pack=True)
        )
    ]
))

page.save()
```

**Reply by Alex-Wenner-FHR** [reply]
Thank you this is super helpful. I am trying to isolate a specific value in one of the columns. Would I just specify the color range and domain that specific value? Expecting it to color that data point different than the others?
